src/utils/shouldRedirect.ts

```diff
@@ -97,7 +99,7 @@ export const shouldRedirect = (
       ['Profile.tsx', { isSelf: P.boolean, name: P.string }],
       (_params) => _params && router.isReady,
       (_params) => {
-        const { name, isSelf, decodedName, normalisedName } = _params[1]
+        const { name, isSelf, decodedName, normalisedName, tabs, tab } = _params[1]
 
         if (
           name !== decodedName &&
```

Comment:
Let's add the rerouting code to all of the conditions here. e.g.

const hasValidTab = tabs.includes(tab)
const tabQuery = tab !== 'profile' && hasValidTab ? `?tab=${tab}` : ''

... .... 
router.replace(`${destination}/${decodedName}${tabQuery})


... ...
if (!hasValidTab) router.replace(...)

Reply:
I encountered an infinite loop when attempting to reroute the valid tab query using the shouldRedirect method. To resolve this, I temporarily commented out the `return router.replace(destination/{name}${tabQuery})` code. The issue is resolved when removing the `router` dependency. I would appreciate your guidance on addressing this infinite loop.